composer/algorithms/algorithm_hparams.py

```diff
@@ -1,64 +1,89 @@
 # Copyright 2022 MosaicML Composer authors
 # SPDX-License-Identifier: Apache-2.0
 
-from __future__ import annotations
-
 import os
-from abc import ABC, abstractmethod
-from dataclasses import dataclass
-from typing import Optional
+from typing import Dict, List, Optional, Type, Union
 
 import yahp as hp
 
 import composer
+from composer.algorithms.agc import AGC
+from composer.algorithms.alibi import Alibi
+from composer.algorithms.augmix import AugMix
+from composer.algorithms.blurpool import BlurPool
+from composer.algorithms.channels_last import ChannelsLast
+from composer.algorithms.colout import ColOut
+from composer.algorithms.cutmix import CutMix
+from composer.algorithms.cutout import CutOut
+from composer.algorithms.ema import EMA
+from composer.algorithms.factorize import Factorize
+from composer.algorithms.ghost_batchnorm import GhostBatchNorm
+from composer.algorithms.label_smoothing import LabelSmoothing
+from composer.algorithms.layer_freezing import LayerFreezing
+from composer.algorithms.mixup import MixUp
+from composer.algorithms.no_op_model import NoOpModel
+from composer.algorithms.progressive_resizing import ProgressiveResizing
+from composer.algorithms.randaugment import RandAugment
+from composer.algorithms.sam import SAM
+from composer.algorithms.selective_backprop import SelectiveBackprop
+from composer.algorithms.seq_length_warmup import SeqLengthWarmup
+from composer.algorithms.squeeze_excite import SqueezeExcite
+from composer.algorithms.stochastic_depth import StochasticDepth
+from composer.algorithms.swa import SWA
 from composer.core.algorithm import Algorithm
 
+algorithm_registry: Dict[str, Union[Type[Algorithm], Type[hp.Hparams]]] = {
```

Comment:
The registry is only for yahp, so I would like to leave hparams or the like in there (e.g. the logger_registry contains the `ObjectStoreLoggerHparams` since that logger has a different hparams path than the functional path). Thoughts about `x_hparams_registry.py` ...would that be too long?

Reply:
Renamed to `x_hparams_registry.py`